src/ingestion.py

```diff
@@ -0,0 +1,24 @@
+from gremlin_connect.gremlin_adapter import GremlinAdapter
+
+from src.graph_model import Traversel
+from src.ingestion_data import IngestionData
+
+gremlin_adapter = GremlinAdapter()
+def _ingest_pcve(pcve):
+    g = Traversel()
+    # (fixme) create ecosystem node if not present
+    # (fixme) use ecosystem node as well
+    # n.create_ecosystem_node(ecosystem_name=pcve['ecosystem'])
+    query = str(g.addV(pcve.dependency).as_('dependency')
+     .addV(pcve.version).as_('version')
```

Comment:
It's always better to do a scalability testing initially with first cut of code. I may be wrong, but, to me it looks like we are over-engineering the oops here.

Reply:
Actually I had implemented a simple Graph Traversal abstraction which outputs query as a string rather than bytecode to connect with Gremlin HTTP. It would be helpful to move to WebSocket based gremlin in future.